AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m

```diff
@@ -1784,31 +1784,42 @@ - (void)testStartUpdateWhenApplicationEnterForeground {
   id notificationCenterMock = OCMPartialMock([NSNotificationCenter new]);
   OCMStub([notificationCenterMock defaultCenter]).andReturn(notificationCenterMock);
   id distributeMock = OCMPartialMock([MSDistribute new]);
-  __block int startUpdateCounter = 0;
-  OCMStub([distributeMock startUpdate]).andDo(^(__attribute((unused)) NSInvocation *invocation) {
-    startUpdateCounter++;
-  });
+  OCMStub([distributeMock startUpdate]).andDo(nil);
+  
+  // When
+  id appCenterMock = OCMClassMock([MSAppCenter class]);
+  OCMStub([appCenterMock sharedInstance]).andReturn(appCenterMock);
+  OCMStub([appCenterMock sdkConfigured]).andReturn(YES);
+  OCMStub([appCenterMock isConfigured]).andReturn(YES);
+  [distributeMock startWithChannelGroup:OCMProtocolMock(@protocol(MSChannelGroupProtocol))
+                              appSecret:kMSTestAppSecret
+                transmissionTargetToken:nil
+                        fromApplication:YES];
+  
+  // Then
+  OCMVerify([distributeMock isEnabled]);
+  OCMVerify([distributeMock startUpdate]);
 
   // When
   [distributeMock setEnabled:NO];
   [notificationCenterMock postNotificationName:UIApplicationWillEnterForegroundNotification object:nil];
 
   // Then
   OCMVerify([distributeMock isEnabled]);
-  XCTAssertEqual(startUpdateCounter, 0);
+  OCMReject([distributeMock startUpdate]);
 
   // When
   [distributeMock setEnabled:YES];
 
   // Then
-  XCTAssertEqual(startUpdateCounter, 1);
+  OCMVerify([distributeMock startUpdate]);
 
   // When
   [notificationCenterMock postNotificationName:UIApplicationWillEnterForegroundNotification object:nil];
 
   // Then
   OCMVerify([distributeMock isEnabled]);
-  XCTAssertEqual(startUpdateCounter, 2);
+  OCMVerify([distributeMock startUpdate]);
```

Comment:
You are missing the count of calls here. If it is called only once, the test will pass